control-plane/agents/src/bin/core/controller/io_engine/v1/translation.rs

```diff
@@ -208,7 +208,8 @@ impl TryIoEngineToAgent for v1::replica::ReplicaSnapshot {
                 .timestamp
                 .clone()
                 .and_then(|t| std::time::SystemTime::try_from(t).ok())
-                .unwrap_or(UNIX_EPOCH),
+                // only valid on create, not on list! should we handle this as None ?
+                .unwrap_or(std::time::SystemTime::now()),
```

Comment:
If we haven't got a valid timestamp in ReplicaSnapshot as part of create snapshot grpc response, then is it right to mark it as now() in control-plane, which looks like a legitimate timestamp?

Reply:
No, but UNIX_EPOCH doesn't seem right either, which is why I've added a comment there; 
we might need to set timestamp as option on our side as well...

Reply:
I reverted back for UNIX_EPOCH for now, but I think we should probably handle this properly rather than fake it..

Reply:
sure, will modify this in a follow-up change.
Unrelated to this directly, but with a placeholder timestamp (UNIX_EPOCH or None) what'll be the interpretation on control-plane about this snapshot, is it anyway useful? Should `list` operation enumerate this, if yes how would a user decide if a clone should be created using this snapshot since we don't know when was this created?

Reply:
One thing we can perhaps do is, as soon as we create a new replica snapshot with valid timestamp, delete any existing snapshot for that replica having placeholder timestamp.

Reply:
CSI says timestamp is REQUIRED so we can't simply make it optional...
Also, it seems that dataplane always returns the timestamp... perhaps this is just an artifact of the grpc protobuf which causes timestamp to be an Option.. so maybe we can keep this unwrap to epoch since it should never happen.